lib/osf-components/addon/components/file-browser/add-new/upload/template.hbs

```diff
@@ -0,0 +1,85 @@
+
+{{#let (unique-id 'upload-files-dropzone') as |id|}}
+    <DropzoneWidget
+        @buildUrl={{this.buildUrl}}
+        @options={{this.dropzoneOptions}}
+        @dropzone={{false}}
+        @addedfile={{action this.addedFile}}
+        @success={{action this.success}}
+        @error={{action this.error}}
+        @enable={{@enable}}
+        @id={{id}}
+        @clickable={{this.clickableElementSelectors}}
+        as |dropzone|
+    >
+        {{yield (hash
+            setClickableElementId=this.setClickableElementId
+        )}}
+        <OsfDialog
+            @isOpen={{this.shouldShowModal}}
+            @onClose={{this.closeModal}}
+        as |dialog|>
+            <dialog.heading>
+                {{#if this.shouldShowUploadingModal}}
+                    {{t 'osf-components.file-browser.uploading_file' fileCount=this.uploading.length}}
+                {{/if}}
+                {{#if this.shouldShowFailureModal}}
+                    {{t 'osf-components.file-browser.upload_failed' fileCount=this.uploadErrored.length}}
+                {{/if}}
+                {{#if this.shouldShowSuccessModal}}
+                    {{t 'osf-components.file-browser.upload_success' fileCount=this.uploadCompleted.length}}
+                {{/if}}
+            </dialog.heading>
+            <dialog.main>
+                {{#if this.shouldShowUploadingModal}}
+                    <ul local-class='FileList'>
+                        {{#each this.uploading as |item|}}
+                            <li local-class='FileListItem'>
+                                <div>
+                                    <FaIcon @icon='file-alt' />
+                                    <span local-class='Filename'>{{item.name}}</span>
+                                </div>
+                                <FaIcon @icon='spinner' @pulse={{true}}/>
```

Comment:
Can we DRY this a little bit by adding `this.modalItems` (name TBD) that fetches the appropriate list based on  `this.shouldShowXYZModal` and iterating through that list? Unless we anticipate some more potential changes to this modal based on uploading/failure/success states, this seems a bit repetitive

Reply:
Uploading, failure and successful modals each have different behaviors regarding the buttons/icons. I don't think it is worth it to DRY this up. This is deliberate to make it more readable so that one does not need to go back to `component.js` to understand what this modal is doing.